### PR TITLE
BUG: Modal for Adding Courses on Basic Course Search Page Displays All Schedules

### LINKED ISSUE #1: BUG: Add course dialog allows adding course to the wrong schedule
### Summary

When using the button on the main course search results to add a course to a personal schedule, the button allows the user to add courses to a schedule for a different quarter than the course.  This should not be allowed. Instead, the modal that shows a list of personal schedules to choose from *should only* show personal schedules for the quarter that matches the course being added.  If no such personal schedules exist, the model should say, for example (if the quarter is F24): 

>No personal schedules exist for quarter F24.
> `[Create One]`

Where `[Create One]` is a button that navigates to the create page for personal schedules.

### Expected Behavior

When adding a course from a given quarter to a personal schedule, only schedules that match the quarter of the course should be presented as options.

### Current / Observed Behavior

When adding a course from a given quarter to a personal schedule, all personal schedules are presented as options. This results in incorrect behavior, where one of two things happens:
* Either there is a POST error when the enrollment code doesn't exist in the target quarter
* A course with the target enrollment code (e.g. 23456) is added, but it could be an *entirely different course* from the one intended.

### Steps to Reproduce

1. Login to the https://courses.dokku-00.cs.ucsb.edu website
2. Create at least one personal schedule for a given quarter (e.g. W22 as shown below)
    
    image

4. Search for a course from a *different* quarter (e.g. S24)
    
    image

6. Find a course you can add with the Green Add button.  Click the button to add that course to a personal schedule.

   
8. The modal pops up as shown below.  Select a schedule for the wrong quarter.
   
    image

10. Observe that you get either an error or you get a course other than the one you intended showing up on your schedule.  
    
Here's what the error looks like:
       
image

Here's what a wrong course looks like. This happens when by coincidence, there is a course with the same enrollment
code in the target quarter (but it could be an entirely different course!  In this case, we tried to add ANTH 3 from F24 but ended up with ANTH 5 from W22.)

image

image
  
image

    

# Acceptance Criteria

- [x] When you click the add button for a course in a given quarter, say F24, in the modal that pops up, all of the personal schedules for that quarter are shown, and ONLY the personal schedules for that quarter are shown.  
- [x] When you click the add button for a course in a given quarter, say F24, if there are no personal schedules for that quarter that already exist, the model says: "There are no personal schedules for F24" (or whatever the quarter is), and there is a button [Create Personal Schedule] that when you click it, navigates to the `https://courses-qa.dokku-00.cs.ucsb.edu/personalschedules/create` page.

### PR DESCRIPTION
**Pull request by andersonlee3**
- fixed the bug so that the modal only displays schedules for the corresponding quarter
- when no schedules match, displays a message saying no schedules exist for the quarter the user is searching on and link to create a new schedule

dokku: [https://al-proj-courses.dokku-04.cs.ucsb.edu](url)
Closes #1

### CODE REVIEW

**Review by andrew344c — DISMISSED** [review]
LGTM

File: frontend/src/main/components/PersonalSchedules/PersonalScheduleSelector.js [hunk 1]
```diff
@@ -1,38 +1,28 @@
...
 const PersonalScheduleSelector = ({
+  filteredSchedules,
   schedule,
   setSchedule,
   controlId,
   onChange = null,
   label = "Schedule",
 }) => {
-  const localSearchSchedule = localStorage.getItem(controlId);
+  //const localSearchSchedule = localStorage.getItem(controlId);
```
Comment (pconrad) [inline frontend/src/main/components/PersonalSchedules/PersonalScheduleSelector.js:14]: Commented out code: 
https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

File: frontend/src/main/components/PersonalSchedules/PersonalScheduleSelector.js [hunk 2]
```diff
@@ -1,38 +1,28 @@
...
 
   const [scheduleState, setScheduleState] = useState(
-    localSearchSchedule || schedule,
+    schedule,
+    //localSearchSchedule || schedule,
```
Comment (pconrad) [inline frontend/src/main/components/PersonalSchedules/PersonalScheduleSelector.js:18]: https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

File: frontend/src/main/components/PersonalSchedules/PersonalScheduleSelector.js [hunk 3]
```diff
@@ -1,38 +1,28 @@
 import React, { useState, useEffect } from "react";
 import { Form } from "react-bootstrap";
-import { useBackend } from "main/utils/useBackend";
+//import { useBackend } from "main/utils/useBackend";
```
Comment (pconrad) [inline frontend/src/main/components/PersonalSchedules/PersonalScheduleSelector.js:3]: https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

**Review by pconrad — CHANGES_REQUESTED** [review]
https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

File: frontend/src/main/components/PersonalSchedules/PersonalScheduleSelector.js (hunk 2 above)
Comment (andersonlee3) [reply frontend/src/main/components/PersonalSchedules/PersonalScheduleSelector.js]: done.

File: frontend/src/main/components/PersonalSchedules/PersonalScheduleSelector.js (hunk 3 above)
Comment (andersonlee3) [reply frontend/src/main/components/PersonalSchedules/PersonalScheduleSelector.js]: done.

**Review by andrew344c — APPROVED** [review]
LGTM

I don't think I see any commented out code

**Pull request merged**